### DISCUSSION TITLE
Separate specific instruments?

### DISCUSSION THREAD

**Discussion by markapeters** [intel/openvino-plugins-ai-audacity#123] [Ideas]
Your plugin is amazing as it is, but can the ability to separate specific/individual instruments be added? For example, as a guitarist (amateur) I'd love to be able to separate all the instruments into individual tracks so that I can record my own version of each guitar track and then mix everything down into one file.

**Comment by gblong1** [comment]
As a musician, myself, I fully agree! However, we need to find an AI model which has this capability. If you hear of something that can do the separation of the individual guitars, please let us know and we can investigate how to integrate it.

Thanks!

**Comment by MarcoRavich** [comment]
Hi everyone, yup it would be extremely useful (for remix/mastering audio) to have all instruments separated in different steams !

To perform this other models are needed.

A nice list of them can be found here:
https://mvsep.com/

Hope that helps/inspires !

**Reply by moderatly** [reply]
Hey Marco, have you used mvsep and if so how were the results for the stem splitting?

**Reply by MarcoRavich** [reply]
> Hey Marco, have you used mvsep and if so how were the results for the stem splitting?

No, since it's an online (and pay-per-use) service, but they have clear list of - mostly free - [models you can grab](https://github.com/ZFTurbo/Music-Source-Separation-Training/blob/main/docs/pretrained_models.md#pre-trained-models) and try yourself.

Anyway I hope @RyanMetcalfeInt8 will add an integrated "model manager" in OpenVINO PFA.

**Comment by robsanderson69** [comment]
I fully endorse this suggestion. Unfortunately I've get to come across an AI model which does anything beyond splitting up a piece of music into anything other than the four usual stems.
If the user was able to pick a small section, say one note, of an instrument for the AI model to sample and be trained on, would that help the AI model to isolate it from the piece of music it was in?
Also, with regard to music separation in general. It would be very useful to be able to train the AI to refine its' separation model, as sometimes the results don't always sound that great.
Having the ability to train the AI model on a small section (just a few seconds) of the song and then pick out the best result for full separation or further refinement would be really useful!

**Reply by MarcoRavich** [reply]
Keep in mind that these plugins aren't trainable and works exclusively by using pretrained models, so you have to find a model that fit your needs (anyway I've never heard of a "separate by sample" kind of model).

**Reply by ghost** [reply]
There's a few recent experimental project toward this like https://github.com/kwatcharasupat/query-bandit/ , text queried source separation is also starting to be a serious thing https://arxiv.org/pdf/2506.16833

**Reply by MarcoRavich** [reply]
> There's a few recent experimental project toward this like https://github.com/kwatcharasupat/query-bandit/ , text queried source separation is also starting to be a serious thing https://arxiv.org/pdf/2506.16833
>
Interesting, it claims _"On the MoisesDB dataset, Banquet, at only 24.9 M trainable parameters, approached the performance level of the significantly more complex 6-stem Hybrid Transformer Demucs on VDBO stems and outperformed it on guitar and piano. The query-based setup allows for the separation of narrow instrument classes such as clean acoustic guitars, and can be successfully applied to the extraction of less common stems such as reeds and organs."_

...have you tried [available models](https://zenodo.org/records/13694558) ? 

Anyway, since @kwatcharasupat is _focusing on audio source separation_, many interesting "uncommon" models has been published so far:
- [BandIt: Cinematic Audio Source Separation](https://github.com/kwatcharasupat/bandit#readme);
- [Bandit Model (Re)implementation](https://github.com/kwatcharasupat/bandit-v2#readme);
- [Divide and Remaster v3](https://github.com/kwatcharasupat/divide-and-remaster-v3#readme);

Hope @RyanMetcalfeInt8's will evaluate to support them too.

**Reply by MarcoRavich** [reply]
Bump.

Final users demand for cinematic separation models is growing (mainly to generate dub "remasters": put an old dub over new HQ audio).

Hope upcoming version will support @kwatcharasupat's [Bandit-v2](https://github.com/kwatcharasupat/bandit-v2#readme) ([weights](https://zenodo.org/records/12701995)) too.
